model/map-jpa/src/main/resources/META-INF/single-use-objects/jpa-single-use-objects-changelog-1.xml

```diff
@@ -0,0 +1,76 @@
+-->
+
+
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd
+                    http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd">
+
+    <changeSet author="keycloak" id="single-use-object-1">
+
+        <createTable tableName="kc_single_use_obj">
+            <column name="id" type="UUID">
+                <constraints primaryKey="true" nullable="false"/>
+            </column>
+            <column name="version" type="INTEGER" defaultValueNumeric="0">
+                <constraints nullable="false"/>
+            </column>
+            <column name="metadata" type="json"/>
+        </createTable>
+        <ext:addGeneratedColumn tableName="kc_single_use_obj">
+            <ext:column name="entityversion" type="INTEGER" jsonColumn="metadata" jsonProperty="entityVersion"/>
+            <ext:column name="objectkey" type="VARCHAR(255)" jsonColumn="metadata" jsonProperty="fObjectKey"/>
+            <ext:column name="expiration" type="BIGINT" jsonColumn="metadata" jsonProperty="fExpiration"/>
+        </ext:addGeneratedColumn>
+        <createIndex tableName="kc_single_use_obj" indexName="single_use_obj_entityVersion">
+            <column name="entityversion"/>
+        </createIndex>
+        <createIndex tableName="kc_single_use_obj" indexName="single_use_obj_objectKey">
+            <column name="objectkey"/>
+        </createIndex>
+        <createIndex tableName="kc_single_use_obj" indexName="single_use_obj_expiration">
+            <column name="expiration"/>
+        </createIndex>
+        <ext:createJsonIndex tableName="kc_single_use_obj" indexName="kc_single_use_obj_nonce">
+            <ext:column jsonColumn="metadata" jsonProperty="fActionVerificationNonce"/>
+        </ext:createJsonIndex>
+
+        <createTable tableName="kc_single_use_obj_note">
+            <column name="id" type="UUID">
+                <constraints primaryKey="true" nullable="false"/>
+            </column>
+            <column name="fk_root" type="UUID">
+                <constraints foreignKeyName="single_use_obj_note_fk_root_fkey" references="kc_single_use_obj(id)" deleteCascade="true"/>
+            </column>
+            <column name="name" type="VARCHAR(255)"/>
+            <column name="value" type="TEXT"/>
+        </createTable>
+        <createIndex tableName="kc_single_use_obj_note" indexName="single_use_obj_note_fk_root">
+            <column name="fk_root"/>
+        </createIndex>
+        <createIndex tableName="kc_single_use_obj_note" indexName="single_use_obj_note_name_value">
+            <column name="name"/>
+            <column name="VALUE(255)" valueComputed="VALUE(255)"/>
```

Comment:
What is this index for? Do we ever search by note's value?

Reply:
no, we don't. Will remove it

Reply:
Done!

Reply:
Thanks, resolved!